src/portfft/descriptor.hpp

```diff
@@ -385,19 +385,19 @@ class committed_descriptor {
         IdxGlobal factor_wi = factor_size / factor_sg;
         if (detail::can_cast_safely<IdxGlobal, Idx>(factor_sg) && detail::can_cast_safely<IdxGlobal, Idx>(factor_wi)) {
           if (batch_interleaved_layout) {
-            // (local_memory_for_input + local_mem_for_store_modifier + local_mem_for_twiddles) < local_mem
-            return (sizeof(Scalar) *
-                    (num_scalars_in_local_mem<detail::layout::BATCH_INTERLEAVED>(
-                         detail::level::SUBGROUP, static_cast<std::size_t>(factor_size), SubgroupSize,
-                         {static_cast<Idx>(factor_sg), static_cast<Idx>(factor_wi)}, temp_num_sgs_in_wg) +
-                     static_cast<std::size_t>(factor_size) * LocalRange + 2 * static_cast<std::size_t>(factor_size))) <
+            return (2 *
+                        num_scalars_in_local_mem<detail::layout::BATCH_INTERLEAVED>(
+                            detail::level::SUBGROUP, static_cast<std::size_t>(factor_size), SubgroupSize,
+                            {static_cast<Idx>(factor_sg), static_cast<Idx>(factor_wi)}, temp_num_sgs_in_wg) *
+                        sizeof(Scalar) +
+                    2 * static_cast<std::size_t>(factor_size) * sizeof(Scalar)) <
                    static_cast<std::size_t>(local_memory_size);
           }
-          return (sizeof(Scalar) *
-                  (num_scalars_in_local_mem<detail::layout::PACKED>(
-                       detail::level::SUBGROUP, static_cast<std::size_t>(factor_size), SubgroupSize,
-                       {static_cast<Idx>(factor_sg), static_cast<Idx>(factor_wi)}, temp_num_sgs_in_wg) +
-                   static_cast<std::size_t>(factor_size) * LocalRange + 2 * static_cast<std::size_t>(factor_size))) <
+          return (num_scalars_in_local_mem<detail::layout::PACKED>(
+                      detail::level::SUBGROUP, static_cast<std::size_t>(factor_size), SubgroupSize,
+                      {static_cast<Idx>(factor_sg), static_cast<Idx>(factor_wi)}, temp_num_sgs_in_wg) *
+                      sizeof(Scalar) +
+                  2 * static_cast<std::size_t>(factor_size) * sizeof(Scalar)) <
```

Comment:
No, this is for the PACKED case, where no store modifiers will be applied

Reply:
Does the code need to change [here](https://github.com/codeplaysoftware/portFFT/blob/7041405af37fb08c7bfae6e7936091336d0ba4da/src/portfft/common/global.hpp#L477) then or am I missing something else?

Reply:
No, notice that assignment of local memory required for modifiers is under a check of whether or not its the last factor, where if its not returns the appropriate local memory size required, or 1 otherwise.